src/pkt_proc.c

```diff
@@ -988,17 +988,27 @@ void* process_packet (unsigned char *ctx_ptr,
     }
 
     ip_len = ntohs(ip->ip_len);
-    if (ip_len < sizeof(struct ip_hdr) || ip_len > header->caplen) {
+    if (ip_len < sizeof(struct ip_hdr)) {
         /*
-         * IP packet is malformed (shorter than a complete IP header, or
-         * claims to be longer than it is), or not entirely captured by
-         * libpcap (which will depend on MTU and SNAPLEN; you can change
-         * the latter if need be).
+         * IP packet is malformed (shorter than a complete IP header)
          */
         if (allocated_packet_header)
             free(dyn_header);
         return NULL;
     }
+    if (ip_len > header->caplen) {
+        /*
+         * IP packet is truncated (claims to be longer than
+         * what was capture by libpcap).
+         * This will depend on MTU and SNAPLEN; you can change
+         * the latter if need be.
+         * Let's reset the ip_len to the length of the caplen minus
+         * the ethernet header and then process the truncated packet.
+         */
+        joy_log_debug("Truncated IP packet: orig len %u , new len %u", ip_len, (header->caplen-14));
```

Comment:
Can we use sizeof(ether_header) for 14 ?

Reply:
add a #define for the length